### PR TITLE
Added support for HF pretrained models. Fixed the docs.

### PR DESCRIPTION
**Pull request by VahidooX**
+Fixed some typo mistakes in the ASR LM docs.
+ Added the support of using HF pretrained models like Transformer-XL and GPT instead of a nemo pretrained model.
User may pass the name of a pretrianed HF model instead of the model path.

### CODE REVIEW

**Review by titu1994 — COMMENTED** [review]
Looks great! Minor comments

File: nemo/collections/nlp/modules/common/huggingface/bert.py [hunk 1]
```diff
@@ -28,6 +28,6 @@ class BertEncoder(BertModel, BertModule):
     """
 
     @typecheck()
-    def forward(self, input_ids, attention_mask, token_type_ids):
+    def forward(self, input_ids, attention_mask=None, token_type_ids=None):
```
[the remark this answers is not in the document]
Comment (VahidooX) [reply nemo/collections/nlp/modules/common/huggingface/bert.py:31]: For types they are already optional. They just missed to add None here.

File: scripts/asr_language_modeling/neural_rescorer/eval_neural_rescorer.py [hunk 2]
```diff
@@ -212,6 +243,7 @@ def autocast():
                 input_ids, input_mask, acoustic_score, dist, ref_len, len_in_chars, idx = batch
 
                 max_len_in_batch = input_mask.sum(dim=0).argmin().item()
+                # max_len_in_batch = len_in_chars.max()
```
[the remark this answers is not in the document]
Comment (VahidooX) [reply scripts/asr_language_modeling/neural_rescorer/eval_neural_rescorer.py:246]: dropped it.

File: scripts/asr_language_modeling/neural_rescorer/eval_neural_rescorer.py [hunk 3]
```diff
@@ -220,9 +252,16 @@ def autocast():
                         acoustic_score.to(device),
                         len_in_chars.to(device),
                     )
+                # some models like Transformer-XL don't need attention_mask as input
+                if "attention_mask" in inspect.getfullargspec(model.forward).args:
+                    log_probs = model(input_ids=input_ids, attention_mask=input_mask)
```
[the remark this answers is not in the document]
Comment (VahidooX) [reply scripts/asr_language_modeling/neural_rescorer/eval_neural_rescorer.py:260]: Done.

**Comment by titu1994** [comment]
LGTM, final review from @AlexGrinch

File: scripts/asr_language_modeling/neural_rescorer/eval_neural_rescorer.py [hunk 4]
```diff
@@ -77,16 +80,29 @@ def __init__(self, data_path, tokenizer, manifest_path, beam_size=128, max_seq_l
         self.beam_size = beam_size
         self.max_seq_length = max_seq_length
 
+        if self.tokenizer.pad_id is not None:
+            self.pad_id = self.tokenizer.pad_id
+        elif self.tokenizer.eos_id is not None:
+            self.pad_id = self.tokenizer.eos_id
+        else:
+            logging.warning(f"Using 0 as pad_id as the tokenizer has no pad_id or eod_id.")
```
Comment (AlexGrinch) [inline scripts/asr_language_modeling/neural_rescorer/eval_neural_rescorer.py:88]: typo: eod_id -> eos_id

File: docs/source/asr/asr_language_modeling.rst [hunk 5]
```diff
@@ -202,14 +202,16 @@ This score is usually combined with the scores from the beam search decoding to
 Train Neural Rescorer
 =====================
 
-An example script to train such a language model with Transformer can be found at `examples/nlp/language_modeling/transformer_lm.py <https://github.com/NVIDIA/NeMo/blob/stable/examples/nlp/language_modeling/transformer_lm.py>`__.
+An example script to train a language model with Transformer can be found at `examples/nlp/language_modeling/transformer_lm.py <https://github.com/NVIDIA/NeMo/blob/stable/examples/nlp/language_modeling/transformer_lm.py>`__.
 It trains a TransformerLMModel which can be used as a neural rescorer for an ASR system.
+You may also use a pretrained language model from HuggingFace (HF) library like Transformer-XL and GPT instead of training your model.
+Model like BERT and RoBERTa are not supported by this script as they are trained as a Masked Language Model and are not efficient and effective to score sentences out of the box.
```
Comment (AlexGrinch) [inline docs/source/asr/asr_language_modeling.rst:208]: typo: Model -> Models

**Review by AlexGrinch — APPROVED** [review]
Some typos in the docs/comments. Also, asr_language_modeling docs should be revised to resolve the conflict with LM training text added recently.
Other than that, LGTM!

**Comment by okuchaiev** [comment]
@VahidooX can you please resolve merge conflicts?

**Pull request merged**